frame/contracts/src/wasm/prepare.rs

```diff
@@ -409,157 +381,97 @@ where
 
-/// Same as [`prepare`] but without constructing a new module.
-///
-/// Used to update the code of an existing module to the newest [`Schedule`] version.
-/// Stictly speaking is not necessary to check the existing code before reinstrumenting because
-/// it can't change in the meantime. However, since we recently switched the validation library
-/// we want to re-validate to weed out any bugs that were lurking in the old version.
-pub fn reinstrument<E, T>(
-	original_code: &[u8],
-	schedule: &Schedule<T>,
-	determinism: Determinism,
-) -> Result<Vec<u8>, DispatchError>
-where
-	E: Environment<()>,
-	T: Config,
-{
-	instrument::<E, T>(
-		original_code,
-		schedule,
-		determinism,
-		// This function was triggered by an interaction with an existing contract code
-		// that will try to instantiate anyways. Failing here would not help
-		// as the contract is already on chain.
-		TryInstantiate::Skip,
-		InstrumentReason::Reinstrument,
-	)
-	.map_err(|(err, msg)| {
-		log::error!(target: LOG_TARGET, "CodeRejected during reinstrument: {}", msg);
-		err
-	})
-	.map(|(code, _)| code)
+	Ok(WasmBlob { code, code_info })
 }
 
 /// Alternate (possibly unsafe) preparation functions used only for benchmarking and testing.
 ///
 /// For benchmarking we need to construct special contracts that might not pass our
-/// sanity checks or need to skip instrumentation for correct results. We hide functions
-/// allowing this behind a feature that is only set during benchmarking or testing to
-/// prevent usage in production code.
+/// sanity checks. We hide functions allowing this behind a feature that is only set during
+/// benchmarking or testing to prevent usage in production code.
 #[cfg(any(test, feature = "runtime-benchmarks"))]
 pub mod benchmarking {
 	use super::*;
 
-	/// Prepare function that neither checks nor instruments the passed in code.
+	/// Prepare function that does not perform most checks on the passed in code.
 	pub fn prepare<T: Config>(
-		original_code: Vec<u8>,
+		code: Vec<u8>,
 		schedule: &Schedule<T>,
 		owner: AccountIdOf<T>,
-	) -> Result<PrefabWasmModule<T>, &'static str> {
-		let contract_module = ContractModule::new(&original_code, schedule)?;
-		let memory_limits = get_memory_limits(contract_module.scan_imports(&[])?, schedule)?;
-		Ok(PrefabWasmModule {
-			instruction_weights_version: schedule.instruction_weights.version,
-			initial: memory_limits.0,
-			maximum: memory_limits.1,
-			code_hash: T::Hashing::hash(&original_code),
-			original_code: Some(original_code.try_into().map_err(|_| "Original code too large")?),
-			code: contract_module
-				.into_wasm_code()?
-				.try_into()
-				.map_err(|_| "Instrumented code too large")?,
-			owner_info: Some(OwnerInfo {
-				owner,
-				// this is a helper function for benchmarking which skips deposit collection
-				deposit: Default::default(),
-				refcount: 0,
-			}),
+	) -> Result<WasmBlob<T>, &'static str> {
```

Comment:
can't we use  Error<T> for the result here (and use <Error<T>>::CodeTooLarge, below)